src/components/materialization/SourceCapture/AddSourceCaptureToSpecButton.tsx

```diff
@@ -83,7 +88,10 @@ function AddSourceCaptureToSpecButton({ toggle }: AddCollectionDialogCTAProps) {
             if (nameUpdated) {
                 setSourceCapture(updatedSourceCapture.capture);
 
-                if (selectedRow?.writes_to) {
+                if (
+                    selectedRow?.writes_to &&
+                    selectedRow?.writes_to.length > 0
```

Comment:
Since we now run a lot more code based on this - we need to make sure the length is there.